### PR TITLE
Implement Coinbase API support

### PR DESCRIPTION
**Pull request by anson-vandoren**
Mainly added to get fiat/BTC prices (instead of needing to grab a 400MB csv file with lots of unneeded data.

### CODE REVIEW

File: ccgains/historic_data.py [hunk 1]
```diff
@@ -415,3 +415,232 @@ def prepare_request(self, dtime):
...
+                self.unit = self.cto + '/' + self.cfrom
+                self.currency_pair = flipped_currency_pair
+                self.file_name = flipped_file_name
+
+    def _wait_if_needed(self):
+        # Wait for minimum call time to pass:
+        now = pd.Timestamp.now()
+        delta = (now - self.last_query_time).total_seconds()
+
+        log.info(f'Coinbase last_query_time: {self.last_query_time}')
```
Comment (probstj) [inline ccgains/historic_data.py]: This seems very verbose, logging every one of Coinbase's query's times. Did you maybe just put it here while debugging and forgot to remove it?

File: ccgains/historic_data.py [hunk 2]
```diff
@@ -415,3 +415,232 @@ def prepare_request(self, dtime):
 
             store.put(key, self.data, format="fixed")
             return self.data
+
+
+class HistoricDataAPICoinbase(HistoricData):
+    last_query_time = pd.Timestamp.now()
+
+    def __init__(self, cache_folder, unit, interval='H'):
+        """Initialize a HistoricData object which tranparently fetches data
```
Comment (probstj) [inline ccgains/historic_data.py]: "... which will *tran**s**parently* fetch ..." No need to repeat my stupid mistakes. :-p

File: ccgains/historic_data.py [hunk 3]
```diff
@@ -415,3 +415,232 @@ def prepare_request(self, dtime):
...
+        except requests.ConnectionError:
+            raise self.connection_error
+
+        # Check for valid response:
+        if response.status_code in [429, 418]:
+            raise ConnectionError('Coinbase Rate limits exceeded')
+        elif response.status_code >= 400:
+            err_msg = response.json()['message']
+            raise ValueError(
+                'Cannot retrieve trade data from Binance '
```
Comment (probstj) [inline ccgains/historic_data.py]: Coinbase ;)

File: ccgains/historic_data.py (hunk 1 above)
Comment (anson-vandoren) [reply ccgains/historic_data.py]: Yup, just a missed debugging statement. Thanks. Will fix this and the others tonight and resubmit

File: ccgains/historic_data.py (hunk 2 above)
Comment (anson-vandoren) [reply ccgains/historic_data.py]: heh, yup, copy-paste got me there :)

File: ccgains/historic_data.py (hunk 3 above)
Comment (anson-vandoren) [reply ccgains/historic_data.py]: Thanks. Will have a new commit fixing all these hopefully tonight

**Comment by anson-vandoren** [comment]
Done. Thanks again!

**Pull request merged**